beacon-chain/core/peerdas/reconstruction.go

```diff
@@ -28,6 +28,56 @@ func MinimumColumnCountToReconstruct() uint64 {
 	return (params.BeaconConfig().NumberOfColumns + 1) / 2
 }
 
+// recoverCellsForBlobs reconstructs cells for all blobs from the given data column sidecars.
+// When withProofs is true, it returns both cells and proofs. When false, it only returns cells (optimized).
+func recoverCellsForBlobs(verifiedRoSidecars []blocks.VerifiedRODataColumn, blobCount int, withProofs bool) ([][]kzg.Cell, [][]kzg.Proof, error) {
```

Comment:
Instead of having `recoverCellsForBlobs` which actually recovers cells and proofs if the `withProofs` argument is `true`, I would favor to have 2 differents functions:
- `recoverCellsForBlobs`
- `recoverCellsAndProofsForBlobs`

(even if it introduces a little code duplication)